cmd/agent/daemon/state/netflow_pipeline.go

```diff
@@ -262,65 +299,16 @@ func (c *Controller) getPodInfo(podID string) (*kubepb.Pod, bool) {
 	return pod, true
 }
 
-func (c *Controller) cleanupNetflow() {
-	c.netflowsMu.Lock()
-	defer c.netflowsMu.Unlock()
-
-	now := time.Now()
-	var totalRemoved int
-	for key, flow := range c.netflows {
-		lastFlowUpdate := now.Sub(flow.updatedAt)
-		if lastFlowUpdate >= c.cfg.NetflowExportInterval*2 {
-			totalRemoved++
-			delete(c.netflows, key)
+func (c *Controller) getConntrackDest(src, dst netip.AddrPort) (netip.AddrPort, bool) {
+	realDst, found := c.ctCache.Get(src)
+	if !found {
+		if realDst, found := c.ct.GetDestination(src, dst); found {
+			c.ctCache.Add(src, realDst)
```

Comment:
Can the destination of an IP change? 
I guess it can and it will, as kubernetes services should be round robin, meaning the ClusterIP will probably point to different real IPs for requests. Or am I wrong here?

Reply:
Cache key is based on source tuple which includes ip+port. For each new connection port will be different. But it will be reused at for different targets at some point for sure.

What do you think if I change ct to have expiration? The issue is that last time I did profile it was quite expensive and creates memory allocations.

Reply:
Mhm maybe a tuple of source + dst would be more resilient as the cache key (I did not check if we have all the info in all places we need though). Just src sounds a bit dangerous, as the same tuple will get reused for a completely different target, which means we would attribute the traffic wrong. 

Not sure how likely this happens though (I have no idea how exactly the kernel determines which port to use for src).